instrumentation/ruby_kafka/lib/opentelemetry/instrumentation/ruby_kafka/patches/producer.rb

```diff
@@ -17,9 +17,14 @@ def produce(value, topic:, key: nil, headers: {}, partition: nil, partition_key:
               'messaging.destination_kind' => 'topic'
             }
 
-            tracer.in_span("#{topic} send", attributes: attributes, kind: :producer) do
-              OpenTelemetry.propagation.inject(headers)
-              super
+            # If trace context is present in headers, extract and use it as parent. The spec mandates that extract
+            # return the current context (Context.current) in Ruby, so this is a noop if propagation headers are absent.
+            ctx = OpenTelemetry.propagation.extract(headers)
```

Comment:
Now I'm second guessing. 

Is the _synchronous_ producer injecting the context into the as well? 

Is there no case where it should use the thread local context?

Reply:
Comment covers this case (perhaps not totally clearly).

```ruby
ctx = OpenTelemetry.propagation.extract(headers)
```

is equivalent to:
```ruby
ctx = OpenTelemetry.propagation.extract(headers, context: Context.current)
```

On the sync `produce` path, unless the caller explicitly passed in `headers` that match the propagation format (there's a test for this case), the `headers` won't contain any trace context, so `ctx` will be `Context.current` after this call. Which means the `OpenTelemetry::Context.with_current(ctx)` below is a no-op - the `Context.current` will be whatever it was before the `extract` above.

It would be nice to have a test that makes this explicit.

Reply:
Yeah, this implementation is perhaps a bit of inside baseball. I will add a test and also try to clarify the comment.

Reply:
I've added the missing test here 1c6ba39688b705f050127163383447f8d1245ed9

Reply:
@arielvalentin lmk if you want to re- 👀 before merge! Thanks for all your attention on the back-and-forth!